src/um/d3d12.rs

```diff
@@ -1638,260 +1792,606 @@ STRUCT!{struct D3D12_STATIC_SAMPLER_DESC {
-    BufferLocation: D3D12_GPU_VIRTUAL_ADDRESS,
-    SizeInBytes: UINT,
-    StrideInBytes: UINT,
+RIDL!{#[uuid(0x6102dee4, 0xaf59, 0x4b09, 0xb9, 0x99, 0xb4, 0x4d, 0x73, 0xf0, 0x9b, 0x24)]
+interface ID3D12CommandAllocator(ID3D12CommandAllocatorVtbl): 
+    ID3D12Pageable(ID3D12PageableVtbl) {
+    fn Reset() -> HRESULT,
 }}
 
+RIDL!{#[uuid(0x0a753dcf, 0xc4d8, 0x4b91, 0xad, 0xf6, 0xbe, 0x5a, 0x60, 0xd9, 0x5a, 0x76)]
+interface ID3D12Fence(ID3D12FenceVtbl): ID3D12Pageable(ID3D12PageableVtbl) {
+    fn GetCompletedValue() -> UINT64,
+    fn SetEventOnCompletion(
+        Value: UINT64,
+        hEvent: HANDLE,
+    ) -> HRESULT,
+    fn Signal(Value: UINT64,) -> HRESULT,
+}}
 
+RIDL!{#[uuid(0x765a30f3, 0xf624, 0x4c6f, 0xa8, 0x28, 0xac, 0xe9, 0x48, 0x62, 0x24, 0x45)]
+interface ID3D12PipelineState(ID3D12PipelineStateVtbl): 
+    ID3D12Pageable(ID3D12PageableVtbl) {
+    fn GetCachedBlob(ppBlob: *mut *mut ID3DBlob,) -> HRESULT,
+}}
 
-RIDL!(#[uuid(0x6102dee4, 0xaf59, 0x4b09, 0xb9, 0x99, 0xb4, 0x4d, 0x73, 0xf0, 0x9b, 0x24)]
-interface ID3D12CommandAllocator(ID3D12CommandAllocatorVtbl): ID3D12Pageable(ID3D12PageableVtbl) {
-    fn Reset() -> HRESULT,
-});
+RIDL!{#[uuid(0x8efb471d, 0x616c, 0x4f49, 0x90, 0xf7, 0x12, 0x7b, 0xb7, 0x63, 0xfa, 0x51)]
+interface ID3D12DescriptorHeap(ID3D12DescriptorHeapVtbl): 
+    ID3D12Pageable(ID3D12PageableVtbl) {
+    fn GetDesc(__ret_val: *mut D3D12_DESCRIPTOR_HEAP_DESC,) -> *mut D3D12_DESCRIPTOR_HEAP_DESC,
+    fn GetCPUDescriptorHandleForHeapStart(__ret_val: *mut D3D12_CPU_DESCRIPTOR_HANDLE,) -> *mut D3D12_CPU_DESCRIPTOR_HANDLE,
+    fn GetGPUDescriptorHandleForHeapStart(__ret_val: *mut D3D12_GPU_DESCRIPTOR_HANDLE,) -> *mut D3D12_GPU_DESCRIPTOR_HANDLE,
+}}
 
-RIDL!(#[uuid(0x7116d91c, 0xe7e4, 0x47ce, 0xb8, 0xc6, 0xec, 0x81, 0x68, 0xf4, 0x37, 0xe5)]
-interface ID3D12CommandList(ID3D12CommandListVtbl): ID3D12DeviceChild(ID3D12DeviceChildVtbl) {
+RIDL!{#[uuid(0x0d9658ae, 0xed45, 0x469e, 0xa6, 0x1d, 0x97, 0x0e, 0xc5, 0x83, 0xca, 0xb4)]
+interface ID3D12QueryHeap(ID3D12QueryHeapVtbl): ID3D12Pageable(ID3D12PageableVtbl) {
+}}
+
+RIDL!{#[uuid(0xc36a797c, 0xec80, 0x4f0a, 0x89, 0x85, 0xa7, 0xb2, 0x47, 0x50, 0x82, 0xd1)]
+interface ID3D12CommandSignature(ID3D12CommandSignatureVtbl): 
+    ID3D12Pageable(ID3D12PageableVtbl) {
+}}
+
+RIDL!{#[uuid(0x7116d91c, 0xe7e4, 0x47ce, 0xb8, 0xc6, 0xec, 0x81, 0x68, 0xf4, 0x37, 0xe5)]
+interface ID3D12CommandList(ID3D12CommandListVtbl): 
+    ID3D12DeviceChild(ID3D12DeviceChildVtbl) {
     fn GetType() -> D3D12_COMMAND_LIST_TYPE,
-});
+}}
+
+RIDL!{#[uuid(0x5b160d0f, 0xac1b, 0x4185, 0x8b, 0xa8, 0xb3, 0xae, 0x42, 0xa5, 0xa4, 0x55)]
+interface ID3D12GraphicsCommandList(ID3D12GraphicsCommandListVtbl): 
+    ID3D12CommandList(ID3D12CommandListVtbl) {
+    fn Close() -> HRESULT,
+    fn Reset(
+        pAllocator: *mut ID3D12CommandAllocator,
+        pInitialState: *mut ID3D12PipelineState,
+    ) -> HRESULT,
+    fn ClearState(pPipelineState: *mut ID3D12PipelineState,) -> (),
```

Comment:
Semantically it should me *mut. 'Reset' must be able to change data hidden behind interface pointer. Also, there are pointer-to-const-data parameters, annotated as _In_ too.  Preserving this distinction is safe bet I guess.